### PR TITLE
Remove Window API that will be questionable in SDL3

### PR DESCRIPTION
**Pull request by Starbuck5**
For #2603, see https://github.com/pygame-community/pygame-ce/issues/2603#issuecomment-1862312329 for details

### CODE REVIEW

**Review by yunline — APPROVED** [review]
LGTM👍

File: src_c/window.c [hunk 1]
```diff
@@ -712,16 +712,6 @@ window_get_opacity(pgWindowObject *self, void *v)
...
 
-static PyObject *
-window_get_display_index(pgWindowObject *self, PyObject *_null)
-{
-    int index = SDL_GetWindowDisplayIndex(self->_win);
-    if (index < 0) {
-        return RAISE(pgExc_SDLError, SDL_GetError());
-    }
-    return PyLong_FromLong(index);
-}
```
Comment (ankith26) [inline src_c/window.c:723]: I'm not clear on why this is being removed, from my understanding SDL3 will continue to provide this info?

Comment (Starbuck5) [reply src_c/window.c:723]: It won’t. A display index and a display id are not the same.

Comment (ankith26) [reply src_c/window.c:723]: Oh, interesting. Thoughts on renaming this to `display_id`, and exporting display id on SDL3 and display index on SDL2? Will this cause confusion or create problems?

Comment (Starbuck5) [reply src_c/window.c:723]: Display id is not numerical. It’s a type. You can use it to look up things about displays.

It’s competently different from an index.

Comment (ankith26) [reply src_c/window.c:723]: Okay, I agree we can just remove it for now and later decide if we want to re-add it in some form

**Review by ankith26 — APPROVED** [review]
LGTM, lets get this in! :100:

**Pull request merged**